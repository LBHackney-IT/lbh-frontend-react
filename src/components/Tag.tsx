import PropTypes from "prop-types";
import classnames from "classnames";
import React from "react";

import { Attributes, DataAttributes } from "../helpers/Attributes";

import "lbh-frontend/lbh/components/lbh-tag/_tag.scss";

/**
 * The proptypes for {@link Tag}.
 *
 * This also supports all `aria-*` and `data-*` attributes.
 *
 * @noInheritDoc
 */
export interface TagProps extends React.AriaAttributes, DataAttributes {
  id?: string;
  className?: string;
  children: React.ReactNode;
}

/**
 * A component for indicating the status of something, such as
 * an item on a task list or a phase banner.
 */
export const Tag: React.FunctionComponent<TagProps> = (
  props: TagProps
): React.ReactElement => {
  const { id, className, children } = props;

  const extraAttributes = Attributes.ariaAndData(props);

  return (
    <strong
      id={id}
      className={classnames("govuk-tag", "lbh-tag", className)}
      {...extraAttributes}
    >
      {children}
    </strong>
  );
};

Tag.propTypes = {
  id: PropTypes.string,
  className: PropTypes.string,
  children: PropTypes.node.isRequired,
};
