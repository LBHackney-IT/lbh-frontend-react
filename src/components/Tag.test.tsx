import React from "react";
import { create } from "react-test-renderer";

import { Tag } from "./Tag";

it("renders correctly without optional props", () => {
  const component = create(
    <Tag aria-label="test tag" data-test="test data">
      Testchildren
    </Tag>
  );

  expect(component).toMatchInlineSnapshot(`
    <strong
      aria-label="test tag"
      className="govuk-tag lbh-tag"
      data-test="test data"
    >
      Testchildren
    </strong>
  `);
});

it("renders correctly with all props", () => {
  const component = create(
    <Tag id="Testid" className="class1 class2">
      <span>Test</span>
    </Tag>
  );

  expect(component).toMatchInlineSnapshot(`
    <strong
      className="govuk-tag lbh-tag class1 class2"
      id="Testid"
    >
      <span>
        Test
      </span>
    </strong>
  `);
});
