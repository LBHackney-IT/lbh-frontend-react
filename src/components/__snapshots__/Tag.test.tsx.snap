// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with all props 1`] = `
<strong
  className="govuk-tag lbh-tag class1 class2"
  id="Testid"
>
  <span>
    Test
  </span>
</strong>
`;

exports[`renders correctly without optional props 1`] = `
<strong
  className="govuk-tag lbh-tag"
>
  Testchildren
</strong>
`;
